static/app/components/compactSelect/listBox/option.tsx

```diff
@@ -42,35 +42,52 @@ export function ListBoxOption({item, listState, size}: ListBoxOptionProps) {
     ref
   );
 
-  const checkboxSize = size === 'xs' ? 'xs' : 'sm';
+  const optionPropsMemo = useMemo(
```

Comment:
Is this necessary to memoize? It gets spread onto MenuListItem so I'd think that the object reference changing wouldn't matter

Reply:
I think so! `optionProps` contains a bunch of event callbacks (`onPointerDown`, `onKeyDown`,…), each of which gets updated with a different reference every time the selection/focus state changes, regardless of whether the current option is affected by that change. So, without memoizing `optionProps`, every time we hover on a new option, all `MenuListItem`s (even the version wrapped inside `memo()`) in the list box will rerender.

Reply:
This wouldn't be necessary if `react-aria`'s `useOption` returned memoized props/callbacks, but it doesn't sadly.